server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ExportService.java

```diff
@@ -45,6 +45,25 @@
 
 public class ExportService {
 
+    public enum Stage {
+        STARTING,
+        ZIPPING,
+        EXPORTING,
+        COMPLETED,
+    }
+
+    public static class Progress {
+        private Stage stage;
+
+        public Stage getStage() {
+            return stage;
+        }
+
+        public void setStage(Stage stage) {
+            this.stage = stage;
+        }
```

Comment:
I prefer to have it immutable.

I agree that it would make the code harder to write, but it will be more explicit.

Reply:
In this case if the Stage variable is declared final then i see no way of updating it later throughout the export process. Do we have to use a different mechanism for tracking progress then? 

If you have any suggestion that would be helpful.

Reply:
How about having `Progress` collecting all stages, storing them in a muttable concurrent friendly collection.

Then when we call 'getStage' it returns the most advanced stage that was notified?

Reply:
@blackheaven i've added some changes following Benoit's suggestion, WDYT?